daal4py/sklearn/model_selection/split.py

```diff
@@ -0,0 +1,201 @@
+        else:
+            if mkl_random_is_imported and (isinstance(random_state, int) or random_state is None):
+                random_state = mkl_random.RandomState(random_state, rng)
+                indexes = random_state.permutation(n_train + n_test)
+                train, test = indexes[:n_train], indexes[n_train:]
+            else:
+                cv = ShuffleSplit(test_size=n_test,
+                                  train_size=n_train,
+                                  random_state=random_state)
+                train, test = next(cv.split(X=arrays[0], y=stratify))
+
+    res = []
+    for arr in arrays:
+        fallback = False
+        if isinstance(arr, np.ndarray):
+            origin_type = "numpy"
+        elif pandas_is_imported:
+            if isinstance(arr, pd.core.frame.DataFrame):
+                origin_type = "pandas DataFrame"
+            elif isinstance(arr, pd.core.series.Series):
+                origin_type = "pandas Series"
+            else:
+                fallback = True
+        else:
+            fallback = True
+
+        dtypes = get_dtypes(arr)
+        if dtypes is None:
+            fallback = True
+        else:
+            for i, dtype in enumerate(dtypes):
+                if dtype in [np.int32, 'int32']:
+                    dtypes[i] = 0
+                elif dtype in [np.float32, 'float32']:
+                    dtypes[i] = 1
+                elif dtype in [np.float64, 'float64']:
+                    dtypes[i] = 2
+                else:
+                    fallback = True
+                    break
+        if fallback:
+            train = train.reshape((n_train,))
+            test = test.reshape((n_test,))
+            res.append(_safe_indexing(arr, train))
+            res.append(_safe_indexing(arr, test))
+        else:
+            train = train.reshape(n_train, 1)
+            test = test.reshape(n_test, 1)
+
+            if len(arr.shape) == 2:
+                n_cols = arr.shape[1]
+                reshape_later = False
+            else:
+                n_cols = 1
+                reshape_later = True
+
+            unique_dtypes = set(dtypes)
+            dtypes_list = dtypes
+            if origin_type == 'pandas DataFrame':
+                if len(unique_dtypes) > 1:
+                    origin_type += ' with multiple dtypes'
+                else:
+                    origin_type += ' with one dtype'
+            if len(dtypes) == 1:
+                dtypes = dtypes * n_cols
+            dtypes = np.array(dtypes, dtype=np.int32, ndmin=2)
+
+            if origin_type == 'numpy':
+                order = 'C' if arr.flags['C_CONTIGUOUS'] == True else 'F'
+
+                train_arr = np.empty(shape=(n_train, n_cols), dtype=arr.dtype, order=order)
+                test_arr = np.empty(shape=(n_test, n_cols), dtype=arr.dtype, order=order)
+
+                d4p.daal_train_test_split(arr.reshape((arr.shape[0], n_cols), order='A'), train_arr, test_arr, train, test, dtypes)

```

Comment:
Why you use reshape? I don't understand why do you create 5 branches for detect input types? `data_or_file` can in inside defined need types for NumericTable.

Reply:
reshaping is mostly free operation in this case and allows to pass data without problem (data_or_file doesn't consume 1d input)